integration/kubernetes/confidential/sev.bats

```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env bats
+# Copyright 2022 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+export TEST_DIR
+export ENCRYPTION_KEY
+
+export KBS_DIR
+export KBS_DB_HOST
+export KBS_DB_USER="kbsuser"
+export KBS_DB_PW="kbspassword"
+export KBS_DB="simple_kbs"
+export KBS_DB_TYPE="mysql"
+
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+NC='\033[0m'
+
+TESTS_REPO_DIR=$(realpath "$BATS_TEST_DIRNAME/../../..")
+test_tag="[cc][kubernetes][containerd][sev]"
+
+esudo() {
+  sudo -E PATH=$PATH "$@"
+}
+
+run_kbs() {
+  KBS_DIR="$(mktemp -d /tmp/kbs.XXXXXXXX)"
+  pushd "${KBS_DIR}"
+
+  # Clone and run
+  git clone https://github.com/confidential-containers/simple-kbs.git --branch main
+  (cd simple-kbs && git checkout -b branch_0.1.1 0.1.1)
+  (cd simple-kbs && esudo docker-compose up -d)
+  sleep 5
+
+  # Set KBS_DB_HOST to kbs db container IP
+  KBS_DB_HOST=$(esudo docker network inspect simple-kbs_default \
+    | jq -r '.[].Containers[] | select(.Name | test("simple-kbs_db.*")).IPv4Address' \
+    | sed "s|/.*$||g")
+
+  popd
+}
+
+pull_encrypted_image_and_set_keys() {
+  # Pull encrypted docker image - test workload
+  esudo docker pull quay.io/kata-containers/encrypted-image-tests:encrypted
+
+  # Get encryption key from docker image label
+  ENCRYPTION_KEY=$(esudo docker inspect quay.io/kata-containers/encrypted-image-tests:encrypted \
+    | jq -r '.[0].Config.Labels.enc_key')
+
+  # Get ssh key from docker image label and save to file
+  esudo docker inspect quay.io/kata-containers/encrypted-image-tests:encrypted \
+    | jq -r '.[0].Config.Labels.ssh_key' \
+    | sed "s|\(-----BEGIN OPENSSH PRIVATE KEY-----\)|\1\n|g" \
+    | sed "s|\(-----END OPENSSH PRIVATE KEY-----\)|\n\1|g" \
+    > ${TEST_DIR}/encrypted-image-tests
+
+  # Set permissions on private key file
+  chmod 600 ${TEST_DIR}/encrypted-image-tests
+}
+
+generate_firmware_measurement_with_append() {
+local append_default="tsc=reliable no_timer_check rcupdate.rcu_expedited=1 \
+i8042.direct=1 i8042.dumbkbd=1 i8042.nopnp=1 i8042.noaux=1 noreplace-smp reboot=k \
+cryptomgr.notests net.ifnames=0 pci=lastbus=0 console=hvc0 console=hvc1 quiet panic=1 \
+nr_cpus=1 scsi_mod.scan=none agent.config_file=/etc/agent-config.toml"
+
+  # Gather firmware locations and kernel append for measurement
+  local append=${1:-${append_default}}
+  local kernel_path="$(esudo kata-runtime kata-env --json | jq -r .Kernel.Path)"
+  local initrd_path="$(esudo kata-runtime kata-env --json | jq -r .Initrd.Path)"
+
```

Comment:
I would check here that `kernel_path` and `initrd_path` exist, unless `sev-snp-measure` fails in case those files doesn't exist. My worry here is to due an internal error it doesn't get the files correctly and `sev-snp-measure` simply ignore that fact then the measurement will be wrong (and it can take time to realize that was wrong).

Reply:
Done.